File: src/java/magmac/app/io/Location.java
package magmac.app.io;

import magmac.api.collect.list.List;
import magmac.api.iter.collect.Joiner;

public record Location(List<String> namespace, String name) {
    @Override
    public String toString() {
        String joined = this.namespace.iter().collect(new Joiner(".")).orElse("");
        return joined + "." + this.name;
    }
}

File: src/web/magmac/app/io/Location.ts
import { List } from "../../../magmac/api/collect/list/List";
import { Joiner } from "../../../magmac/api/iter/collect/Joiner";
export class Location {
	public toString() : String { let joined : var=this.namespace.iter( ).collect( new Joiner( ".")).orElse( "");return joined+"."+this.name;;}
}
